### ISSUE TITLE
PT017 violated in non-test code

### ISSUE THREAD

**Issue by michaelwooley** [m-burst/flake8-pytest-style#255]
# Bug report

## What's wrong

Rule PT017 is incorrectly raised in non-test code.

In particular, it arises if the code:

1. Involves a `try-except` block
2. Within the `except` block, an assertion is made on the caught error

Here is a MWE (with comments so that the flake8 warning is isolated):

```python
"""test.py"""


def f() -> None:
    """docstring"""
    try:
        raise Exception("test")
    except Exception as e:
        assert e is not None # Violation
        assert 1 == 1 # No violation

```

Here is the result of running flake8:

```bash
❯ flake8 test.py
test.py:9:9: PT017 found assertion on exception e in except block, use pytest.raises() instead
```

And a screencap:

![image](https://github.com/m-burst/flake8-pytest-style/assets/8423171/608d5f8e-8c9d-400f-b9aa-3e316a546cb2)

## How it should work

PT017 should not be raised for this case.

I don't think it is ideal to be `assert`-ing in this case but I also don't think that the correct fix is to use `pytest.raises`.

## System information

* Operating system: Ubuntu
* Python version: `3.9.7`
* flake8 version: `6.0.0`
* flake8-pytest-style version: `1.7.2`

**Comment by m-burst** [comment]
Hi @michaelwooley,

Thanks for the report! You are correct, reporting `PT017` doesn't make a lot of sense outside of test code.
However, right now `flake8-pytest-style` doesn't try to distinguish test and non-test code, and it is impossible to do this with 100% correctness.

One easy fix that I can suggest is adding all `PT` violations to [`per-file-ignores`](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-per-file-ignores) for all non-test directories.

**Comment by michaelwooley** [comment]
@m-burst thanks for your reply and work on this package. Sounds good.
